File: Typescript/src/reversePrefix.ts
export function reversePrefix(s: string, k: number): string {
	let arr = s.split('');
	let left = 0;
	let right = k - 1;
	while (left < right) {
		[arr[left], arr[right]] = [arr[right], arr[left]];
		left++;
		right--;
	}
	return arr.join('');
};

File: C/exercise/reversePrefix.c

#include "reversePrefix.h"

char *reversePrefix(char *s, int k)
{
	for (int i = 0; i < (k / 2); i++) {
		char temp = s[i];
		s[i] = s[(k - 1) - i];
		s[(k - 1) - i] = temp;
	}
	return s;
}
